src/pod.ts

```diff
@@ -12,6 +12,17 @@ import {StaticFile} from './static';
 import Cache from './cache';
 import {Locale, LocaleSet} from './locale';
 import {TranslationString, StringOptions} from './string';
+import profiler from './profile';
+
+const storageExistsTimers = profiler.timersFor(
```

Comment:
Is it possible for us to use "start" and "stop" (or "before" and "after") similar to how Eleventy does it?

https://github.com/11ty/eleventy/blob/330191c10b9ab4b93a3abcf6406bbc350a47aa7d/src/TemplatePassthrough.js#L82

One thing I want to be mindful of is the amount of tracing through the code contributors need to do. I think here for a contributor to understand what's going on, they need to grok the location where the code is called, and then trace through the code and come up here to see what the wrapped functions are doing.

I think it'd be really nice and convenient if we had a way to have markers/profilers that we could place around code lines that do the profiling, without having to wrap functions (which may potentially get wrapped again). Thoughts?

Maybe like:

```
profiler.timer(Profile.Timers.ReadFile).start()
... code that reads files ...
profiler.timer(Profile.Timers.ReadFile).stop()
```

That's really clear to me. And then we can create more actionable metrics around `ReadFile` and provide hints to developers in the output, like: "Render templates: XXX minutes. Reduce complexity of templates to improve build time." (or something like that.) I want to be mindful of doing too much magic (i.e. with function wrapping) as it may make the code harder to maintain.

Also should we have the `Pod` hold the profiler so it's not a global singleton? That might be more robust in the future.

```
pod.profiler.timer(Profile.Timers.ReadFile).start()
...etc.
```

Reply:
I've moved the profiler to be part of the pod. And updated the timer to have an explicit `start()` instead of having an automatic start time.

The reason that I am using a `wrap` instead of manually calling the `start` and `stop` for the storage is for performance. By wrapping I only have to look up the timers one time instead of having to look for the timers each time a file operation happens. I think the for normal usage you would do something like:

```ts
const timer = pod.profiler.timer('render.document').start()
// Code that does something...
timer.stop()
```

We could create a `before` and `after` style, but it would make that type of timer a singleton. For example, you could only track one named operation at a time. So if you start a yaml load timer, but then in the yaml loading it tries to do another yaml read timer (for `!a.yaml` style constructor) then it would have trouble because the `before` has already been called for a yaml read.

Reply:
For a `profile.timer()` we could also have it auto start the timer so the code for quick timers would be shorter:

```ts
const timer = pod.profiler.timer('render.document')
// Code that does something...
timer.stop()
```